mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/quota/CassandraPerUserMaxQuotaManagerV2.java

```diff
@@ -0,0 +1,332 @@
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.cassandra.quota;
+
+import static org.apache.james.util.ReactorUtils.publishIfPresent;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.james.backends.cassandra.components.CassandraQuotaLimitDao;
+import org.apache.james.core.Domain;
+import org.apache.james.core.quota.QuotaComponent;
+import org.apache.james.core.quota.QuotaCountLimit;
+import org.apache.james.core.quota.QuotaLimit;
+import org.apache.james.core.quota.QuotaScope;
+import org.apache.james.core.quota.QuotaSizeLimit;
+import org.apache.james.core.quota.QuotaType;
+import org.apache.james.mailbox.model.Quota;
+import org.apache.james.mailbox.model.QuotaRoot;
+import org.apache.james.mailbox.quota.MaxQuotaManager;
+
+import com.google.common.collect.ImmutableMap;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public class CassandraPerUserMaxQuotaManagerV2 implements MaxQuotaManager {
+
+    private static final String GLOBAL_IDENTIFIER = "global";
+
+    private final CassandraQuotaLimitDao cassandraQuotaLimitDao;
+
+    @Inject
+    public CassandraPerUserMaxQuotaManagerV2(CassandraQuotaLimitDao cassandraQuotaLimitDao) {
+        this.cassandraQuotaLimitDao = cassandraQuotaLimitDao;
+    }
+
+    @Override
+    public void setMaxStorage(QuotaRoot quotaRoot, QuotaSizeLimit maxStorageQuota) {
+        setMaxStorageReactive(quotaRoot, maxStorageQuota).block();
+    }
+
+    @Override
+    public Mono<Void> setMaxStorageReactive(QuotaRoot quotaRoot, QuotaSizeLimit maxStorageQuota) {
+        return cassandraQuotaLimitDao.setQuotaLimit(QuotaLimit.builder().quotaScope(QuotaScope.USER).identifier(quotaRoot.getValue())
+            .quotaComponent(QuotaComponent.MAILBOX).quotaType(QuotaType.SIZE).quotaLimit(QuotaCodec.quotaValueToLong(maxStorageQuota)).build());
```

Comment:
Don't chain all build operations on the same line, return to the line for each, like for Reactor chains. Will be easier to read